React Project/src/pages/SignUp/SignUp.tsx

```diff
@@ -0,0 +1,151 @@
+import { useState } from "react";
+import { Link } from "react-router-dom";
+import TextField from "../../components/TextField";
+import Button from "../../components/Button";
+import { encrypt } from "@omar-sarfraz/caesar-cipher";
+import bcrypt from "bcryptjs";
+
+export type SignUpError = {
+    type: "" | "FIRST_NAME" | "LAST_NAME" | "EMAIL" | "PASSWORD" | "CONFIRM_PASSWORD";
+    message: string;
+};
+
+export type User = {
+    firstName: string;
+    lastName: string;
+    email: string;
+    password: string;
+};
+
+export default function SignUp() {
+    const [firstName, setFirstName] = useState<string>("");
+    const [lastName, setLastName] = useState<string>("");
+    const [email, setEmail] = useState<string>("");
+    const [password, setPassword] = useState<string>("");
+    const [confirmPassword, setConfirmPassword] = useState<string>("");
+    const [error, setError] = useState<SignUpError>({ type: "", message: "" });
+
+    const handleSignUp = async () => {
+        setError({ type: "", message: "" });
+
+        if (!firstName) {
+            setError({ type: "FIRST_NAME", message: "First Name is required!" });
+            return;
+        }
+
+        if (!lastName) {
+            setError({ type: "LAST_NAME", message: "Last Name is required!" });
+            return;
+        }
+
+        if (!email) {
+            setError({ type: "EMAIL", message: "Email is required!" });
+            return;
+        }
+
+        if (!password || password.length < 8) {
+            setError({ type: "PASSWORD", message: "Password must be 8 or more characters long!" });
+            return;
+        }
+
+        if (!confirmPassword) {
+            setError({ type: "CONFIRM_PASSWORD", message: "Please confirm your password!" });
+            return;
+        }
+
+        if (password !== confirmPassword) {
+            setError({ type: "CONFIRM_PASSWORD", message: "Passwords do not match!" });
+            return;
+        }
+
+        const key = parseInt(import.meta.env.VITE_CIPHER_KEY);
+
+        const encryptedFirstName = encrypt(key, firstName);
+        const encryptedLastName = encrypt(key, lastName);
+        const encryptedEmail = encrypt(key, email);
+
+        const hash = await bcrypt.genSalt(key);
+        const encryptedPassword = await bcrypt.hash(password, hash);
+
+        let userData: User = {
+            firstName: encryptedFirstName,
+            lastName: encryptedLastName,
+            email: encryptedEmail,
+            password: encryptedPassword,
+        };
+        let existingUser = localStorage.getItem(encryptedEmail);
+
+        if (existingUser) alert("User with this email already exists");
+        else {
+            localStorage.setItem(encryptedEmail, JSON.stringify(userData));
+            alert("Account registered successfully");
```

Comment:
Please create a custom Alert component that is async, the browser alert are intrusive and disliked by users.

Reply:
Alright. I'll add a custom component.